pkg/cyclic/mark.go

```diff
@@ -96,98 +95,56 @@ func (m MarkMap) shouldFilterTxn(startTs uint64, filterReplicaIDs []uint64) (*mo
 	return markRow, false
 }
 
-// MapMarkRowsGroup maps row change events into two group, normal table group
-// and mark table group.
-func MapMarkRowsGroup(
-	inputs map[model.TableName][][]*model.RowChangedEvent,
-) (output TxnMap, markMap MarkMap) {
-	output = make(map[uint64]map[model.TableName][]*model.RowChangedEvent)
-	markMap = make(map[uint64]*model.RowChangedEvent)
-	for name, input := range inputs {
-		for _, events := range input {
-			// Group mark table rows by start ts
-			if IsMarkTable(name.Schema, name.Table) {
-				for _, event := range events {
-					first, ok := markMap[event.StartTs]
-					if ok {
-						// TiKV may emit the same row multiple times.
-						if event.CommitTs != first.CommitTs || event.RowID != first.RowID {
-							log.Fatal(
-								"there should be at most one mark row for each txn",
-								zap.Uint64("start-ts", event.StartTs),
-								zap.Any("first", first),
-								zap.Any("second", event))
-						}
+// FilterAndReduceTxns filters duplicate txns bases on filterReplicaIDs and
+// set the replicaID into txns
+func FilterAndReduceTxns(txnsMap map[model.TableName][]*model.Txn, filterReplicaIDs []uint64, replicaID uint64) {
```

Comment:
Please enhance comments, the function only set replicaID if a txn does not have one.